src/main/java/org/spongepowered/common/registry/type/tileentity/StructureModeRegistryModule.java

```diff
@@ -24,51 +24,30 @@
  */
 package org.spongepowered.common.registry.type.tileentity;
 
-import com.google.common.collect.ImmutableSet;
-import com.google.common.collect.Maps;
 import net.minecraft.tileentity.TileEntityStructure;
 import org.spongepowered.api.data.type.StructureMode;
 import org.spongepowered.api.data.type.StructureModes;
-import org.spongepowered.api.registry.CatalogRegistryModule;
 import org.spongepowered.api.registry.util.AdditionalRegistration;
 import org.spongepowered.api.registry.util.RegisterCatalog;
+import org.spongepowered.common.registry.AbstractCatalogRegistryModule;
 
-import java.util.Collection;
-import java.util.Locale;
-import java.util.Map;
-import java.util.Optional;
-
-import static com.google.common.base.Preconditions.checkNotNull;
-
-public final class StructureModeRegistryModule implements CatalogRegistryModule<StructureMode> {
-
-    @RegisterCatalog(StructureModes.class)
-    public final Map<String, StructureMode> structureModes = Maps.newHashMap();
+@RegisterCatalog(StructureModes.class)
+public final class StructureModeRegistryModule extends AbstractCatalogRegistryModule<StructureMode> {
 
     @Override
     public void registerDefaults() {
-        this.structureModes.put("corner", (StructureMode) (Object) TileEntityStructure.Mode.CORNER);
-        this.structureModes.put("data", (StructureMode) (Object) TileEntityStructure.Mode.DATA);
-        this.structureModes.put("load", (StructureMode) (Object) TileEntityStructure.Mode.LOAD);
-        this.structureModes.put("save", (StructureMode) (Object) TileEntityStructure.Mode.SAVE);
-    }
-
-    @Override
-    public Optional<StructureMode> getById(String id) {
-        return Optional.ofNullable(this.structureModes.get(checkNotNull(id, "id").toLowerCase(Locale.ENGLISH)));
-    }
-
-    @Override
-    public Collection<StructureMode> getAll() {
-        return ImmutableSet.copyOf(this.structureModes.values());
+        this.map.put("corner", (StructureMode) (Object) TileEntityStructure.Mode.CORNER);
```

Comment:
This should also be prefixed with `minecraft:` (although it will be added by default) to match the other registry classes.
(Or it should be removed from all)